### PR TITLE
chore: implement path mutation for JsonFlat

### PR DESCRIPTION
**Pull request by romange**
This is done by converting the flat blob into mutable c++ json and then building a new flat blob.

Also, add JsonFlat encoding to CompactObject class.

### CODE REVIEW

File: src/core/compact_object.h [hunk 1]
```diff
@@ -374,8 +377,12 @@ class CompactObj {
...
   struct JsonWrapper {
-    JsonType* json_ptr = nullptr;
-    size_t unneeded = 0;
+    union {
+      JsonType* json_ptr;
+      uint8_t* flat_ptr;
+    };
+    uint32_t json_len = 0;
+    uint8_t encoding = 0;
   } __attribute__((packed));
```
Comment (BorysTheDev) [inline src/core/compact_object.h:386]: What about ARM?

File: src/core/compact_object.h [hunk 2]
```diff
@@ -18,6 +18,8 @@ constexpr unsigned kEncodingIntSet = 0;
 constexpr unsigned kEncodingStrMap = 1;   // for set/map encodings of strings
 constexpr unsigned kEncodingStrMap2 = 2;  // for set/map encodings of strings using DenseSet
 constexpr unsigned kEncodingListPack = 3;
+constexpr unsigned kEncodingJsonCons = 0;
+constexpr unsigned kEncodingJsonFlat = 1;
```
[the remark this answers is not in the document]
Comment (romange) [reply src/core/compact_object.h:22]: I prefer to keep it as constants to be consistent with other encodings in the system

File: src/core/compact_object.h (hunk 1 above)
Comment (romange) [reply src/core/compact_object.h:386]: what about it?

Comment (BorysTheDev) [reply src/core/compact_object.h:386]: This attribute changes alignment, but in this case it looks do nothing

Comment (BorysTheDev) [reply src/core/compact_object.h:386]: ARM doesn't like incorrect allignment

Comment (romange) [reply src/core/compact_object.h]: move the attribute to more relevant place (line 399) and added a comment.

Comment (romange) [reply src/core/compact_object.h]: We had this code since the inception and it worked fine on arm.

Comment (BorysTheDev) [reply src/core/compact_object.h]: because it does nothing in this case)

Comment (BorysTheDev) [reply src/core/compact_object.h]: The problem can be if you reorder elements

**Review by BorysTheDev — APPROVED** [review]

**Review by BorysTheDev — APPROVED** [review]

**Pull request merged**